src/format.rs

```diff
@@ -974,6 +1061,77 @@ impl<'a> Formatter<'a> {
         }
         .append(self.pattern(&arg.value))
     }
+
+    // TODO: Merge with segment_option() somehow
+    fn pattern_segment(&mut self, segment: &UntypedPatternBinSegment) -> Document {
+        match segment {
+            UntypedPatternBinSegment { value, options, .. } if *options == [] => {
+                self.pattern(&value)
+            }
+
+            UntypedPatternBinSegment { value, options, .. } => {
+                // TODO: Is there a better option than these self.clone() calls?
+                let options = options
+                    .iter()
```

Comment:
Same as above, why is the formatter cloned here?

Reply:
Same